tensorboard/examples/plugins/example_basic/tensorboard_plugin_example/plugin.py

```diff
@@ -92,16 +93,23 @@ def _serve_greetings(self, request):
         """
         run = request.args.get("run")
         tag = request.args.get("tag")
+        ctx = plugin_util.context(request.environ)
+        experiment = plugin_util.experiment_id(request.environ)
+
         if run is None or tag is None:
             raise werkzeug.exceptions.BadRequest("Must specify run and tag")
         try:
-            data = [
-                tensor_util.make_ndarray(event.tensor_proto)
-                .item()
-                .decode("utf-8")
-                for event in self._multiplexer.Tensors(run, tag)
-            ]
+            read_result = self.data_provider.read_tensors(
+                ctx,
+                downsample=1,
+                plugin_name=metadata.PLUGIN_NAME,
+                experiment_id=experiment,
+                run_tag_filter=provider.RunTagFilter(runs=[run], tags=[tag]),
+            )
+
+            data = read_result.get(run, {}).get(tag, {})
```

Comment:
If there is no actual data for this run and tag, the result will be an empty dictionary, but then on the next line `data[0]` will fail.  It's also not actually the right type - to match `read_tensors()`, that type should be a list.

I'd just change it to `.get(tag, [])` and then add an explicit check, like

```py
if not data:
    raise werkzeug.exceptions.BadRequest("Invalid run or tag")
event_data = [data[0].numpy.item().decode("utf-8")]
```

The `except KeyError` below only worked with multiplexer.Tensors(), it won't work with the code you have now since `.get()` on a dict won't actually raise a KeyError.